execution/model/operator.go

```diff
@@ -22,18 +22,26 @@ type OperatorTelemetry interface {
 	IncrementSamplesAtTimestamp(samples int, t int64)
 	Samples() *stats.QuerySamples
 	SubQuery() bool
+	StepInvariant() bool
```

Comment:
It feels odd to use booleans here, is there a situation where subquery and setpinvariant are true at same time? maybe we could just inject the logicalplan.Node here - it probably has all the information ( and more ) we could ever want

Reply:
Does this mean, NewTelemetry should accept an additional parameter for logical node?

```
func NewStepInvariantOperator(...) (model.VectorOperator, error) {
	u := &stepInvariantOperator{
		...
	}
	u.OperatorTelemetry = model.NewTelemetry(u, opts, StepInvariantExpr)
	
}

func NewTelemetry(operator fmt.Stringer, logicalNode Node, opts *query.Options) {
    ...
}

```

Reply:
I followed your suggestion of injecting the logicalplan.Node there. I had to move the refactor telemetry interfaces into its own package to avoid circular dependency.
It does look better.